frameworks/MLNet/exec.py

```diff
@@ -63,8 +63,8 @@ def run(dataset: Dataset, config: TaskConfig):
             columns[label] = 'label'
 
             # might use a bit extra memory, but it's faster
-            pd.DataFrame(dataset.train.data, columns=columns).to_csv(train_dataset_path, index=None)
-            pd.DataFrame(dataset.test.data, columns=columns).to_csv(test_dataset_path, index=None)
+            pd.DataFrame(dataset.train.data_enc, columns=columns).to_csv(train_dataset_path, index=None)
+            pd.DataFrame(dataset.test.data_enc, columns=columns).to_csv(test_dataset_path, index=None)
```

Comment:
can you please explain why you need the encoded version?
I have an ongoing PR that applies the following changes for MLNET: https://github.com/openml/automlbenchmark/pull/293/files#diff-c36c7a5171f7dc3dc648e4143bb874ef5993db7b908ba25263be950d0a503d0d

it simplifies the integration, but the csv file will be as "rich" as possible: it is expected that frameworks that can load a CSV file should be able to load them without having to modify the data before hand.

Reply:
The purpose of using encoding data is to avoid inferring column purpose incorrectly . MLNet uses the first 50 lines of data to infer column purposes. (numeric/text/categorical...) and the rule it uses to determine whether a column should be treated as text or categorical type is the repetition rate. If the distinguished value of that column versus row number below a hard-code threshold then that column will be treated as categorical and one-hot encoder or hash encoder will be applied afterward. Otherwise, text featurizer will be used to that column. The problem is this rule doesn't always work because there's always exceptions. And if exception happens, applying wrong featurizer can be a catastrophe to training results.

For example, in openml/adult, while all columns should be either numeric or categorical column, mlnet incorrectly infers "work class", "education", "martial status", "occupation" as text column, which causes long training time and bad training performance.

A better solution will be loading `.arff` which includes data schema, but unfortunately it's still not supported by ML.Net framework.